### PR TITLE
Feat: Ranking mechanism + Score in each context 

### PR DESCRIPTION
**Pull request by thaohp113**
Problem: for every context in the more information section, it has same answer relevancy score (which does not make sense).

I have fix the context in both UI and backend file. They should be fine by now.

### CODE REVIEW

**Review by Huy1711 — APPROVED** [review]
LGTM

File: app/llama_sensei/backend/qa/generate_answer.py [hunk 1]
```diff
@@ -57,7 +57,10 @@ def __init__(self, course: str, model=MODEL):
         self.embedder = SentenceTransformer(EMBEDDING_LLM, trust_remote_code=True).to(
             DEVICE
         )
-        self.model = ChatGroq(model=model, temperature=0)
+        self.groq_api_key = groq_api_key
+        self.model = ChatGroq(
+            model=model, temperature=0, groq_api_key=self.groq_api_key
```
Comment (khankaholic) [inline app/llama_sensei/backend/qa/generate_answer.py]: @thaohp113 
API key can be accidentally disclosed if loading like your.
Should load it from `.env` and consider to use `python-dotenv` if you don't want to export it manually.

File: tests/unit/backend/vectordb/test_vector_db_operations.py
Comment (khankaholic) [inline tests/unit/backend/vectordb/test_vector_db_operations.py:1]: remove comment if possible

**Review by khankaholic — CHANGES_REQUESTED** [review]

File: tests/unit/backend/vectordb/test_vector_db_operations.py
Comment (Huy1711) [reply tests/unit/backend/vectordb/test_vector_db_operations.py:1]: This test file will be solved in my PR, so please consider ignore this

File: app/llama_sensei/backend/qa/generate_answer.py (hunk 1 above)
Comment (thaohp113) [reply app/llama_sensei/backend/qa/generate_answer.py]: solved by using os.env
`import os
os.environ["GROQ_API_KEY"] = "valid_key"
GROQ_API_KEY = os.environ.get("GROQ_API_KEY")`

**Review by micache — APPROVED** [review]
LGTM

**Pull request merged**